src/core/mac/mac.cpp

```diff
@@ -569,6 +596,12 @@ void Mac::ScheduleNextTransmission(void)
     else
     {
         mState = kStateIdle;
+
+        if (mDisableRadioOnIdle)
+        {
+            mDisableRadioOnIdle = false;
+            DisableRadio();
+        }
```

Comment:
What if we issue scan command (while interface is down), and before it's finished the interface is brought up (which then enables the radio). The `mDisableRadioOnIdle` will remain true, and after scan finishes, we get here and disable the radio.